docs/markdown/IDE-integration.md

```diff
@@ -4,39 +4,110 @@ short-description: Meson's API to integrate Meson support into an IDE
+The `meson-info` directory should contain the following files:
 
-    meson introspect --targets
+ File                            | Description
+ ------------------------------- | ---------------------------------------------------------------------
+ `intro-benchmarks.json`         | Lists all benchmarks
+ `intro-buildoptions.json`       | Contains a full list of meson configuration options for the project
+ `intro-buildsystem_files.json`  | Full list of all meson build files
+ `intro-dependencies.json`       | Lists all dependencies used in the project
+ `intro-installed.json`          | Contains mapping of files to their installed location
+ `intro-projectinfo.json`        | Stores basic information about the project (name, version, etc.)
+ `intro-targets.json`            | Full list of all build targets
+ `intro-tests.json`              | Lists all tests with instructions how to run them
 
-The JSON formats will not be specified in this document. The easiest way of learning them is to look at sample output from the tool.
+The content of the JSON files is further specified in the remainder of this document.
 
-Once you have a list of targets, you probably need the list of source files that comprise the target. To get this list for a target, say `exampletarget`, issue the following command.
+## The `targets` section
 
-    meson introspect --target-files exampletarget
+The most important file for an IDE is probably `intro-targets.json`. Here each
+target with its sources and compiler parameters is specified. The JSON format
+for one target is defined as follows:
 
-In order to make code completion work, you need the compiler flags for each compilation step. Meson does not provide this itself, but the Ninja tool Meson uses to build does provide it. To find out the compile steps necessary to build target foo, issue the following command.
+```json
+{
+    "name": "Name of the target",
+    "id": "The internal ID meson uses",
+    "type": "<TYPE>",
+    "filename": ["list", "of", "generated", "files"],
+    "build_by_default": true / false,
+    "target_sources": [],
+    "installed": true / false,
+}
+```
 
-    ninja -t commands foo
+If the key `installed` is set to `true`, the key `install_filename` will also
+be present. It stores the installation location for each file in `filename`.
+If one file in `filename` is not installed, its corresponding install location
+is set to `null`.
 
-Note that if the target has dependencies (such as generated sources), then the commands for those show up in this list as well, so you need to do some filtering. Alternatively you can grab every command invocation in the [Clang tools db](https://clang.llvm.org/docs/JSONCompilationDatabase.html) format that is written to a file called `compile_commands.json` in the build directory.
```

Comment:
This change removed all mentions of `compile_commands.json` from the docs, even though it's still there.

That led to some confusion: https://github.com/mesonbuild/meson/issues/3545#issuecomment-491765340